### ISSUE TITLE
Can't use currentSelection from unsafeInternalContext because its type is not exposed

### ISSUE THREAD

**Issue by jjant** [morpheusgraphql/morpheus-graphql#492]
I'm trying to traverse the internal context to perform a kind of batched fetching, but I can't do anything with `currentSelection` (other than printing it) because its type is not exposed. Am I missing something?

**Comment by nalchevanidze** [comment]
add `morpheus-graphql-core` to dependecny list and

import `Data.Morpheus.Types.Internal.AST`

**Comment by jjant** [comment]
@nalchevanidze Thanks, I kept looking for an export of those but couldn't find them, my bad!

**Comment by jjant** [comment]
@nalchevanidze I'm running into the same thing with `MergeSet` now:

```
myFunc = do
  Context {currentSelection} <- unsafeInternalContext
  let Selection {selectionContent} = currentSelection
  let (SelectionSet mergeSet) = selectionContent
  -- How to access MergeSet's fields?
```

-----

To give the concrete use case, I have this query:

```
{
  add_ons {
    add_on_id
    name
  }
}
```

Thus, my `currentSelection` at that query's resolver is (gotten through `show`ing `currentSelection` above):

```haskell
Selection
  { selectionName = "add_ons",
    selectionAlias = Nothing,
    selectionPosition = Position {line = 2, column = 3},
    selectionArguments = OrderedMap {mapKeys = [], mapEntries = fromList []},
    selectionContent =
      SelectionSet
        ( MergeSet
            { unpack =
                [ Selection
                    { selectionName = "name",
                      selectionAlias = Nothing,
                      selectionPosition = Position {line = 3, column = 5},
                      selectionArguments = OrderedMap {mapKeys = [], mapEntries = fromList []},
                      selectionContent = SelectionField,
                      selectionDirectives = []
                    },
                  Selection
                    { selectionName = "add_on_id",
                      selectionAlias = Nothing,
                      selectionPosition = Position {line = 4, column = 5},
                      selectionArguments = OrderedMap {mapKeys = [], mapEntries = fromList []},
                      selectionContent = SelectionField,
                      selectionDirectives = []
                    }
                ]
            }
        ),
    selectionDirectives = []
  }
```

I just want to access the queried field names, `["name", "add_on_id" ]`, is there a simpler way than what I'm doing?

**Comment by nalchevanidze** [comment]
Use "Data.Morpheus.Types.SelectionTree"

**Comment by jjant** [comment]
That's unexposed, so I couldn't use it: https://github.com/morpheusgraphql/morpheus-graphql/blob/master/morpheus-graphql-core/morpheus-graphql-core.cabal#L168

**Comment by nalchevanidze** [comment]
oh. thats bad. 
possibly it was deexposed after the spliting to small packages. i will expose it again.

**Comment by nalchevanidze** [comment]
i fixed it in #494

**Comment by nalchevanidze** [comment]
exposed in release 0.14.0
